src/expire.c

```diff
@@ -182,6 +182,12 @@ void activeExpireCycle(int type) {
     long total_sampled = 0;
     long total_expired = 0;
 
+    /* Sanity: There can't be any pending commands to propagate when
+     * we're in cron */
+    serverAssert(server.also_propagate.numops == 0);
+    server.core_propagates = 1;
```

Comment:
btw, is there a reason why "server.core_propagates"  isn't restored before leaving the function, other than the fact the activeExpireCycle isn't expected to be invoked by other commands or in the middle of a script?